[CS2113-W10-1] BudgetBuddy

telferang opened the pull request:
BudgetBuddy is an easy-to-use expense tracking and budgeting tool that helps students stay organised and understand their spending habits. It is optimised for CLI users, allowing faster expense entry by typing in commands.

okkhoy commented on src/main/java/seedu/budgetbuddy/Parser.java:
```diff
@@ -0,0 +1,196 @@
+        case "budget": {
+            YearMonth budgetDate = YearMonth.parse(parts[2], DateTimeFormatter.ofPattern("yyyy-MM"));
+            // Adjust date format for YearMonth
+            String categoryPart = parts[3].trim();
+            categoryPart = categoryPart.substring(1, categoryPart.length() - 1);
```
can be  extracted for better SLAP

okkhoy commented on src/main/java/seedu/budgetbuddy/Storage.java:
```diff
@@ -0,0 +1,183 @@
+
+        assert expenseList != null : "Expense list cannot be null";  // Assert that the expense list is not null
+        assert incomeList != null : "Income list cannot be null";      // Assert that the income list is not null
+        assert budgetList != null : "Budget list cannot be null";      // Assert that the budget list is not null
```
redundant comments here

okkhoy commented on src/main/java/seedu/budgetbuddy/Storage.java:
```diff
@@ -0,0 +1,183 @@
+
+        FileWriter fw = new FileWriter(filePath, false); // Overwrites the file
+
+        // Save expenses
```
redundant comment

remove all such redundant ones

okkhoy commented on src/main/java/seedu/budgetbuddy/commands/budget/DeductBudgetCommand.java:
```diff
@@ -0,0 +1,58 @@
+     */
+    @Override
+    public void execute() {
+        // Assume validation has guaranteed that existingBudget is not null
```
do you think you can use assert here instead of this comment?

okkhoy commented on src/main/java/seedu/budgetbuddy/commands/expense/ListExpenseCommand.java:
```diff
@@ -0,0 +1,99 @@
+        if (category == null && month == null) {
+            LOGGER.log(Level.INFO, "Displaying expenses listed with no Filter");
+            ExpenseManager.listExpenses();
+        } else if (category == null){
```
spacing after the condition

okkhoy commented on src/main/java/seedu/budgetbuddy/graphs/ExpensesOverMonthGraph.java:
```diff
@@ -0,0 +1,89 @@
+        List<Double> xAxis = new ArrayList<>();
+        List<Double> yAxis = new ArrayList<>();
+
+        // Loop through each month of the year
```
redundant comment again

okkhoy commented on src/main/java/seedu/budgetbuddy/transaction/expense/ExpenseManager.java:
```diff
@@ -0,0 +1,347 @@
+                counter++;
+            }
+        }
+        if(result.equals("")) {
```
spacing issues

okkhoy commented on src/main/java/seedu/budgetbuddy/transaction/saving/SavingsManager.java:
```diff
@@ -0,0 +1,118 @@
+        YearMonth firstIncome = YearMonth.now();
+        YearMonth firstExpense = YearMonth.now();
+        String result = "";
+        if (IncomeManager.getIncomes().size() > 0){
+            for (Income income: IncomeManager.getIncomes()){
+                savings += income.getAmount();
+                YearMonth incomeYearMonth = IncomeManager.getYearMonthFromDate(income.getDate());
+                if (incomeYearMonth.compareTo(firstIncome) == -1){
+                    firstIncome = incomeYearMonth;
+                }
+            }
+        }
```
looks deeply nested. can you try to use some guard clause to eliminate the nesting if possible?

okkhoy reviewed (commented):
In general, the code quality is good. There are some minor issues that are easily fixable. Pls take a look at the comments in the review.

Chinorea committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-1/tp

Chinorea committed:
Add Edit Income Feature

Added functions & dependencies for EditIncome feature
Added JavaDoc Comments to EditIncome feature
Added Automated text-ui-tests for EditIncome feature

Chinorea committed:
Merge pull request #108 from Chinorea/branch-Edit-Income

Add Edit Income Function

ZiliaAJY committed:
Add sequence diagram for Add Budget feature in DG

ZiliaAJY committed:
Merge remote-tracking branch 'upstream/master' into branch-DG

Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-1/tp

ZiliaAJY committed:
Update EXPECTED.TXT to include new output format for 'list budget' feature

ZiliaAJY committed:
Merge pull request #109 from ZiliaAJY/branch-DG

Update DG for Add Budget Feature

ZiliaAJY committed:
Update details for Deduct Budget and List Budget features in DG

ZiliaAJY committed:
Add sequence diagram for Deduct Budget feature

- Modified sequence diagram of Add Budget feature

ZiliaAJY committed:
Merge pull request #110 from ZiliaAJY/branch-DG

Update details in Deduct Budget feature for DG

ZiliaAJY committed:
Fix auto-correction of invalid dates

ZiliaAJY committed:
Merge pull request #111 from ZiliaAJY/branch-bugfixes

Fix auto-correction of invalid dates

Chinorea committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-1/tp

Chinorea committed:
Update DG with ListExpense Feature

Chinorea committed:
Update Sequence Diagram for ListExpense

Alfred-Goh02 committed:
Add Function to Display PieChart of Expense

Users need a clear way to visualise their Expenses for a month in
their respective categories.

Add a function with XChart that displays a PieChart which is colour-
coded and shows percentage of Expenses.

The PieChart provide enough information to allow users to make
informed decisions about their expenses.

Alfred-Goh02 committed:
Edit Code For CheckStyle Tests

Alfred-Goh02 committed:
Merge pull request #112 from Alfred-Goh02/branch-piechart

Add Function to Display PieChart of Expense

Chinorea committed:
Update DG for ListIncome Feature

telferang committed:
Update developer guide

-Add addExpense sequence diagram
-Add listRemainingBudget sequence diagram
-Add expense, income and transaction class diagram

telferang committed:
Update user guide

-add expense
-delete expense
-add income
-delete income
-list remaining budget

telferang committed:
Fix bug when "delete income" or "delete expense" is entered

telferang committed:
Fix remove the extra empty line when displaying remaining budget

telferang committed:
Update text ui test

telferang committed:
Add missing new line for some function

telferang committed:
Merge pull request #114 from telferang/branch-UG

Update UG and DG

Chinorea committed:
Update DG for Edit Expenses Feature

Chinorea committed:
Update DG for EditIncome Feature

Chinorea committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-1/tp into branch-DeveloperGuide

# Conflicts:
#	docs/DeveloperGuide.md

Chinorea committed:
Update UG with updated features

Chinorea committed:
Merge pull request #115 from Chinorea/branch-DeveloperGuide

Update DG and UG

Chinorea committed:
Merge branch 'branch-DeveloperGuide'

ZiliaAJY committed:
Allow dates to be optional when adding or deducting budgets

ZiliaAJY committed:
Update UG with Add, Deduct and List Budgets features

ZiliaAJY committed:
Fix UI display and update EXPECTED.TXT to include updated display for budgets

ZiliaAJY committed:
Merge pull request #116 from ZiliaAJY/branch-UG

Update UG

Alfred-Goh02 committed:
Fix PNG links in DG

Alfred-Goh02 committed:
Merge pull request #118 from Alfred-Goh02/branch-DG

Fix PNG links in DG

ryan-txn commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,413 @@
+The methods for `Expense` and `Income` are not shown.
+
+
+![ExpenseAndIncomeClassDiagram.drawio.png](diagrams/ExpenseAndIncomeClassDiagram.drawio.png)
```
If modelling a dependency relation should the arrow be dashed?
![image](https://github.com/user-attachments/assets/96982006-fefb-46f0-9759-bd58c52a62c0)

ryan-txn replied:
Should variables' accessibility (private public etc) be shown in the class diagram once it has been decided? (there is no default accessibility in class diagrams)
![image](https://github.com/user-attachments/assets/23a902d1-31b5-42fd-8ad3-cade5fd9ac7b)

ryan-txn commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,413 @@
+2. Validator also returns different type of Command Classes depending on the validity of input, which has been omitted
+for the same reason as above.
+
+![CommandClass.drawio.png](diagrams/CommandClass.drawio.png)
```
Image could be clearer / use larger font to ensure readability
![image](https://github.com/user-attachments/assets/a2b33f25-7239-4e55-9dec-08a6187eed8b)

ryan-txn commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,413 @@
+
+The following UML Sequence diagram shows how the Parser works to obtain the relevant inputs for the Add Expense Feature:
+The scenario of invalid input and sequence within `RemainingBudgetManager` are omitted to reduce complexity.
+![AddExpenseSequenceDiagram.drawio.png](diagrams/AddExpenseSequenceDiagram.drawio.png)
```
Is alt the right frame to use? Maybe opt should be considered if there is no else path
![image](https://github.com/user-attachments/assets/484bec1e-323d-495d-8bac-0804abb86d4e)

ryan-txn commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,413 @@
+`displayToUser()` method.
+
+Below is a sequence diagram representing the execution of the Search Expense interaction: 
+![SearchExpenseSequenceDiagram.drawio.png](diagrams/SearchExpenseSequenceDiagram.drawio.png)
```
Are alt and opt swapped? Recall alt should contain alternative paths and opt should contain a single optional path.
![image](https://github.com/user-attachments/assets/30aaf7c0-5e03-49c3-8bb0-1c722157bac8)

ryan-txn commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,413 @@
+user using the `Ui` class `displayToUser()` method. 
+
+Below is a sequence diagram representing the execution of the Display Expense interaction:
+![DisplaySavingsSequenceDiagram.drawio.png](diagrams/DisplaySavingsSequenceDiagram.drawio.png)
```
Should the different frames overlap like this? Also I believe again the alt and opt might have been swapped.
![image](https://github.com/user-attachments/assets/5c7e8544-20b6-4e79-b3ee-d73375990f85)

ryan-txn commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,413 @@
+
+The class diagram below indicates the structure of the DisplaySavings Feature, involving `SavingsManager`, `Saving`, 
+`IncomeManager` and `ExpenseManager`.
+![SavingsManagerClassDiagram.drawio.png](diagrams/SavingsManagerClassDiagram.drawio.png)
```
if methods are called by the SavingsManager class, should the class be declared as static and have its name in italics?
![image](https://github.com/user-attachments/assets/7bb849b6-0b2c-4951-86bf-4fe617f2f1b2)

iaso1774 commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,413 @@
+
+The following UML Sequence diagram shows how the Parser works to obtain the relevant inputs 
+for the Add Budget Feature:
+![AddBudgetSequenceDiagram.drawio.png](diagrams/AddBudgetSequenceDiagram.drawio.png)
```
Is writing return on the dashed lines necessary? Should it be blank instead?
![image](https://github.com/user-attachments/assets/8d12c32b-6205-4108-bcf9-d21ddfdc8141)

iaso1774 commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,413 @@
+`processEdit()` function will be called to update values in the Income object.
+
+The following UML Sequence diagram shows how the Parser works to obtain the relevant inputs for the Edit Income Feature
+![EditIncomeSequenceDiagram.drawio.png](diagrams/EditIncomeSequenceDiagram.drawio.png)
```
Should the call to make a new BudgetBuddyException instead be BuddyBudgetException()
![image](https://github.com/user-attachments/assets/663eff9a-38fa-4623-b1f3-bce92a914fb3)

iaso1774 commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,413 @@
+The following UML Sequence diagram shows how the Parser works to obtain the relevant inputs for the List Income Feature
+
+The following UML Sequence diagram shows how the Parser works to obtain the relevant inputs for the List Income Feature
+![ListIncomeSequenceDiagram.drawio.png](diagrams/ListIncomeSequenceDiagram.drawio.png)
```
Should the :IncomeManager class be overlapping with the alt block?
![image](https://github.com/user-attachments/assets/2dd5ab38-4c7e-47df-bae1-46c02213e98b)

iaso1774 replied:
Should the constructor be called here for :ListIncomeCommand?
![image](https://github.com/user-attachments/assets/0e39c4bd-9d55-47c4-b179-715cb0365226)

iaso1774 reviewed (commented):
Overall, these sequence and class diagrams are very easy to follow and don't contain glaring notation errors. Good job!

Alfred-Goh02 committed:
Update UG

Alfred-Goh02 committed:
Fix Text-UI Test

Alfred-Goh02 committed:
Merge pull request #121 from Alfred-Goh02/branch-UG

Update UG

ZiliaAJY committed:
Update input.txt and EXPECTED.TXT for Budget display

ZiliaAJY committed:
Merge pull request #122 from ZiliaAJY/branch-bugfixes

Update input.txt and EXPECTED.TXT for Budget display

Commit:
Remove deep nesting in SavingsManager displaySavingsByMonth method

Commit:
Fix indentation checkstyle error

ChuaCleon committed:
Merge pull request #124 from ChuaCleon/branch_removeDeepNesting

Remove deep nesting in SavingsManager displaySavingsByMonth method

Commit:
Update DisplaySavings and SearchExpense sequence diagrams.
Update DG for DisplayHelp and BreakdownExpense features.
Change isCommand() in HelpCommand to return true for commands starting with "help"

ChuaCleon committed:
Merge pull request #126 from ChuaCleon/branch_updateDGBreakdownExpense

Update DG Sequence diagrams.

Commit:
Edit DG Sequence diagrams.

ChuaCleon committed:
Merge pull request #127 from ChuaCleon/branch_updateDGBreakdownExpense

Edit DG Sequence diagrams.

Alfred-Goh02 committed:
Fix Coding Styles based on PR

Commit:
Add positive JUnit test for breakdownExpensesByCategory

Alfred-Goh02 committed:
Merge pull request #129 from Alfred-Goh02/branch-fixes

Fix Coding Styles based on PR

ChuaCleon committed:
Merge pull request #130 from ChuaCleon/branch-addMoreJUnitTests

Add positive JUnit test for breakdownExpensesByCategory

ZiliaAJY committed:
Add unit tests for Budget-related classes

Commit:
Edit displayHelpMessage() method in Ui class

ZiliaAJY committed:
Merge pull request #131 from ZiliaAJY/branch-JUnit

Add JUnit tests for Budget-related classes

Commit:
Merge branch 'master' into branch-addMoreJUnitTests

ChuaCleon committed:
Merge pull request #132 from ChuaCleon/branch-addMoreJUnitTests

Edit displayHelpMessage in Ui class

Alfred-Goh02 committed:
Add Features to and Reformat DG

Alfred-Goh02 committed:
Merge pull request #133 from Alfred-Goh02/branch-DG

Add Features to and Reformat DG

Alfred-Goh02 committed:
Modify Diagram based on PR Comments

Alfred-Goh02 committed:
Merge pull request #134 from Alfred-Goh02/branch-DG

Modify Diagram based on PR Comments

ZiliaAJY committed:
Add sequence diagrams

- New diagrams for list budget and display income spent
- Update previous diagrams according to comments

ZiliaAJY committed:
Update details for list budget and display income spent in DG

ZiliaAJY committed:
Merge remote-tracking branch 'upstream/master' into branch-DG

ZiliaAJY committed:
Merge pull request #135 from ZiliaAJY/branch-DG

Update DG

Chinorea committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-1/tp

Commit:
Reformat DG to add parts relevant to different subsections
Update UG for new functionalities

ChuaCleon committed:
Merge pull request #136 from ChuaCleon/branch-reformatDG

Reformat DG to add parts relevant to different subsections

telferang committed:
Update developer guide

telferang committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-1/tp into branch-DG-update

# Conflicts:
#	docs/DeveloperGuide.md

telferang committed:
Merge pull request #137 from telferang/branch-DG-update

Developer Guide update

Alfred-Goh02 committed:
Update and Reformat Style on UG

Alfred-Goh02 committed:
Merge pull request #138 from Alfred-Goh02/branch-UG

Update and Reformat Style on UG

Commit:
Add instructions for manual testing of features
Add target user profile and value proposition of BudgetBuddy

ChuaCleon committed:
Merge pull request #140 from ChuaCleon/branch-manualTesting

Add instructions for manual testing of features

Chinorea committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-1/tp

ZiliaAJY committed:
Update details for budgets and display income spent in UG

ZiliaAJY committed:
Arrange order for budgets and display income spent

ZiliaAJY committed:
Merge pull request #141 from ZiliaAJY/branch-UG

Update UG and DG

Chinorea committed:
Update DG Contents (List and Edit Features)

Update List and Edit Feature [Based on updated formatting]

Chinorea committed:
Merge branch 'branch-DeveloperGuide' into branch-Edit-Expense

# Conflicts:
#	data/BudgetBuddy.txt
#	logs/app.log

Chinorea committed:
Align EditCommands to validator style and Fix incorrect output for errornous category input

Add validator to first set of input commands for edit commands
Fix incorrect output to display incorrect message instead of parsing value to others.

Chinorea committed:
Merge branch 'branch-Edit-Expense' into branch-DeveloperGuide

Chinorea committed:
Update DG (Edit Features)

Update DG diagrams and description to align with validator architecture

Chinorea committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-1/tp into branch-DeveloperGuide

# Conflicts:
#	data/BudgetBuddy.txt

Chinorea committed:
Fix checkstyles

Remove unused imports and functions
Fix Automated CLI test ouputs

Chinorea committed:
Fix Checkstyle Errors

Chinorea committed:
Merge pull request #142 from Chinorea/branch-DeveloperGuide

Update DG and Fix Functions Bugs

Alfred-Goh02 committed:
Complete Manual Testing Section in DG and Fix Bug

Alfred-Goh02 committed:
Merge pull request #144 from Alfred-Goh02/branch-DG

Complete Manual Testing Section in DG and Fix Bug

telferang committed:
Update command to list expenses and list incomes

telferang committed:
Merge remote-tracking branch 'tpUpstream/master' into branch-standardize

# Conflicts:
#	src/main/java/seedu/budgetbuddy/validators/expense/ListExpenseValidator.java

telferang committed:
Update expected.txt

telferang committed:
Merge pull request #146 from telferang/branch-standardize

Standardize list command

Chinorea committed:
Update DG

Chinorea committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-1/tp into branch-DeveloperGuide

Chinorea committed:
Merge pull request #147 from Chinorea/branch-DeveloperGuide

Update DG

Commit:
Edit DG sequence diagrams according to comments
Edit displaySavings methods to remove extra newline character.

ChuaCleon committed:
Merge pull request #148 from ChuaCleon/branch-updateDGSequenceDiagramComments

Edit DG sequence diagrams according to comments

Alfred-Goh02 committed:
Fix UML Diagrams for DG

Alfred-Goh02 committed:
Merge pull request #149 from Alfred-Goh02/branch-DG

Fix UML Diagrams for DG

ZiliaAJY committed:
Update manual testing for budgets and display income spent in UG

ZiliaAJY committed:
Modify display of budget deletion

ZiliaAJY committed:
Fix false deduction when specified category does not exist for budget

- Throw exception with message instead

ZiliaAJY committed:
Add BeforeEach for JUnit Tests for DeductBudgetValidatorTest

Alfred-Goh02 committed:
Add Additional Test Cases for Manual Testing

Alfred-Goh02 committed:
Merge pull request #150 from Alfred-Goh02/branch-DG

Add Additional Test Cases for Manual Testing

ZiliaAJY committed:
Add JUnit Tests for Display Income Spent

ZiliaAJY committed:
Merge conflicts with upstream/master and branch-JUnit

ZiliaAJY committed:
Remove unused import

ZiliaAJY committed:
Update EXPECTED.TXT to include updated display of budget deletion

ZiliaAJY committed:
Merge pull request #151 from ZiliaAJY/branch-JUnit

Add JUnit Tests for display income spent

Chinorea committed:
Update JUnit Test

Added JUnit Test for List and Edit feature validators

Chinorea committed:
Merge pull request #152 from Chinorea/master

Update JUnit Test

Alfred-Goh02 committed:
Add Activity Diagram and Edit DG

Alfred-Goh02 committed:
Merge pull request #153 from Alfred-Goh02/branch-DG

Add Activity Diagram and Edit DG

ZiliaAJY committed:
Bold text

ZiliaAJY committed:
Fix saving duplicate budgets to storage

ZiliaAJY committed:
Merge pull request #154 from ZiliaAJY/branch-bugfixes

Fix saving duplicate budgets to storage

ZiliaAJY committed:
Fix Developer Guide alignment

ZiliaAJY committed:
Merge pull request #155 from ZiliaAJY/branch-DG

Fix Developer Guide alignment

Chinorea committed:
Add PPP

Chinorea committed:
Merge pull request #157 from Chinorea/branch-PPP

Add PPP

for #156

ZiliaAJY committed:
Update PPP

ZiliaAJY committed:
Merge pull request #158 from ZiliaAJY/branch-PPP

Update PPP

Commit:
Remove first income and first expense display for DisplaySavings feature.
Replace with Total Income and Total Expense display to user.

ChuaCleon committed:
Merge pull request #214 from ChuaCleon/branch-removeFirstExpenseAndIncome

Remove first income and first expense display for DisplaySavings feature

Commit:
Display all savings numbers in 2 decimal places.

ChuaCleon committed:
Merge pull request #215 from ChuaCleon/branch-removeFirstExpenseAndIncome

Display all savings numbers in 2 decimal places.

Commit:
Edit SearchExpenseCommand and SearchExpenseValidator to search for commands starting with "search expenses"

ChuaCleon committed:
Merge pull request #216 from ChuaCleon/branch-searchExpensesInvalidCommand

Edit SearchExpenseCommand and SearchExpenseValidator

Commit:
Create PPP document

ChuaCleon committed:
Merge pull request #217 from ChuaCleon/branch-createPPP

Create PPP document

Commit:
Edit displayHelpMessage method to display features to new user.

ChuaCleon committed:
Merge pull request #218 from ChuaCleon/branch-editDisplayHelpMessage

Edit displayHelpMessage method to display features to new user.

Commit:
Edit isCommand() method of BreakdownExpensesCommand to validate commands equals to "breakdown expenses" instead of those
starting with "breakdown expenses"

ChuaCleon committed:
Merge pull request #219 from ChuaCleon/branch-updateBreakdownExpenseBug

Edit isCommand() method of BreakdownExpensesCommand

Alfred-Goh02 committed:
Add JUnit Tests to Improve Coverage

1. Add JUnit Tests for expenses commands, validators, storage and parser
2. Fix Issues from Mock PE by Testers

Alfred-Goh02 committed:
Edit Code to Pass CheckStyle Tests

Alfred-Goh02 committed:
Include Braces for If Statements in Storage Test

Alfred-Goh02 committed:
Merge pull request #220 from Alfred-Goh02/branch-J-Unit

Add JUnit Tests to Improve Coverage

Alfred-Goh02 committed:
Change Frame Name for Display Charts

Alfred-Goh02 committed:
Merge pull request #221 from Alfred-Goh02/branch-J-Unit

Change Frame Name for Display Charts

Commit:
Add JUnit test for Expense Manager to improve coverage

ChuaCleon committed:
Merge pull request #222 from ChuaCleon/branch-AddJUnit

Add JUnit test for Expense Manager to improve coverage

Commit:
Update Project Portfolio Page.

ChuaCleon committed:
Merge pull request #223 from ChuaCleon/branch-updatePPP

Update Project Portfolio Page.

Commit:
Change "search expense" to "search expenses" in User Guide.
Remove formatting error in displayHelpMessage method in Ui.

ChuaCleon committed:
Merge pull request #224 from ChuaCleon/branch-editUserGuide

Change "search expense" to "search expenses" in User Guide.

Chinorea committed:
Add Additional Explanation and fix bugs in UG for list and edit commands

Chinorea committed:
Align Parsing of date and month parsing

allow m/ to accept m/3/2024 instead of only m/03/2024

telferang committed:
Add PPP file

Chinorea committed:
Fix formatting of UG information for Edit and List features

telferang committed:
Merge pull request #225 from telferang/branch-ppp

Add PPP file

Chinorea committed:
Fix Edge Cases output bug for Edit features

Chinorea committed:
Fix images not showing on UG

Chinorea committed:
Merge pull request #226 from Chinorea/branch-PE-D-Issues

Fix issues from PE-D

Chinorea committed:
Update PPP and add placeholder image

Chinorea committed:
Update PPP and add placeholder image

Update PPP and add placeholder image

ZiliaAJY committed:
Edit deduct budget description to provide clarity

ZiliaAJY committed:
Make display of budget categories more organised

ZiliaAJY committed:
Add details to deduct budget

ZiliaAJY committed:
Stop tracking logs

ZiliaAJY committed:
Handle unexpected inputs in commands

ZiliaAJY committed:
Restrict input amount to 2dp only

ZiliaAJY committed:
Update listBudgets() to include case of no budgets recorded

ZiliaAJY committed:
Update list budgets display

ZiliaAJY committed:
Update EXPECTED.TXT to include new UI display

ZiliaAJY committed:
Update JUnit Tests

- For budget-related classes

ZiliaAJY committed:
Add images of expected outputs into UG

ZiliaAJY committed:
Update JUnit tests

ZiliaAJY committed:
Merge pull request #228 from ZiliaAJY/branch-PED-issues

Fix PE-D issues

ZiliaAJY committed:
Remove PPP sample file

ZiliaAJY committed:
Update PPP

ZiliaAJY committed:
Merge pull request #229 from ZiliaAJY/branch-PPP

Update PPP

Commit:
Add example usages of features into UG.

telferang committed:
Update PPP

telferang committed:
Update user guide

ChuaCleon committed:
Merge pull request #230 from ChuaCleon/branch-updateUGExample

Add example usages of features into UG.

telferang committed:
Update parser to ignore storage entries with negative amount

telferang committed:
Update the code to show amount with only 2 decimal place

telferang committed:
Update expected.txt

telferang committed:
Add screenshot for user guide

telferang committed:
Merge pull request #231 from telferang/branch-bug

Branch PED bug

Alfred-Goh02 committed:
Complete Personal Portfolio Page

Alfred-Goh02 committed:
Merge pull request #232 from Alfred-Goh02/branch-PPP

Complete Personal Portfolio Page

Alfred-Goh02 committed:
Edit Code to Adhere to Java Coding Standard

Some of the codes authored is not according to coding standard.

This commit fixes Javadoc Comments and Coding standards.

telferang committed:
Refactor amountValidator code

telferang committed:
Merge remote-tracking branch 'tpUpstream/master' into branch-bug

Alfred-Goh02 committed:
Fix Table on AboutUs Page

Alfred-Goh02 committed:
Edit Code for CheckStyle Error

Alfred-Goh02 committed:
Merge pull request #233 from Alfred-Goh02/branch-BugFix

Fix Code Bugs and About Us Page

telferang committed:
Update PPP

telferang committed:
Merge pull request #234 from telferang/branch-bug

Branch Update PPP

Chinorea committed:
Update PPP

Chinorea committed:
Fix PE-D identified issues

Chinorea committed:
Fix automated CLI test expected out

ZiliaAJY committed:
Update PPP

ZiliaAJY committed:
Merge pull request #236 from ZiliaAJY/branch-PPP

Update PPP

Chinorea committed:
Fix Text and JUnit test case values

Chinorea committed:
Fix JUnit Test Cases

Chinorea committed:
Merge pull request #235 from Chinorea/branch-PE-D-Issues

Fix PE-D issues

ZiliaAJY committed:
Update PPP

Alfred-Goh02 committed:
Fix JUnit Tests Bugs and User Guide Bugs

ZiliaAJY committed:
Merge pull request #237 from ZiliaAJY/branch-PPP

Update PPP

Alfred-Goh02 committed:
Merge pull request #238 from Alfred-Goh02/branch-BugFix

Fix JUnit Tests Bugs and User Guide Bugs

Alfred-Goh02 committed:
Update PPP with Relevant Pull Requests

Alfred-Goh02 committed:
Merge pull request #239 from Alfred-Goh02/branch-PPP

Update PPP with Relevant Pull Requests

Alfred-Goh02 committed:
Modify UML Diagram to Fix PED Issues

Alfred-Goh02 committed:
Merge pull request #240 from Alfred-Goh02/branch-DG

Modify UML Diagram to Fix PED Issues

telferang committed:
Refactor code according to coding standard

telferang committed:
Add DeleteIncomeValidator Junit test

telferang committed:
Remove wildcard import

telferang committed:
Merge pull request #241 from telferang/branch-bug

Branch coding standard and Junit

telferang committed:
Update UG

telferang committed:
Merge pull request #242 from telferang/branch-bug

Update UG

Alfred-Goh02 committed:
Fix Bugs and Update UG

Alfred-Goh02 committed:
Merge pull request #243 from Alfred-Goh02/branch-BugFix

Fix Bugs and Update UG

Alfred-Goh02 committed:
Revert "Fix Bugs and Update UG"

telferang committed:
Update PPP

Alfred-Goh02 committed:
Merge pull request #244 from AY2425S1-CS2113-W10-1/revert-243-branch-BugFix

Revert "Fix Bugs and Update UG"

telferang committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-1/tp into branch-updatePPP

telferang committed:
Merge pull request #245 from telferang/branch-updatePPP

Update PPP

Commit:
Update readme to display PPP

Commit:
Merge branch 'branch-updateReadme'

Alfred-Goh02 committed:
Update PPP and UG

Alfred-Goh02 committed:
Fix Validator Test

Alfred-Goh02 committed:
Merge pull request #246 from Alfred-Goh02/branch-PPP

Update PPP and UG

Chinorea committed:
Update Edit feature naming

Chinorea committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-1/tp into branch-PE-D-Issues

Chinorea committed:
Update JUnit Tests

ZiliaAJY committed:
Update Home Page

Chinorea committed:
Merge pull request #247 from Chinorea/branch-PE-D-Issues

Update Naming Convention for Edit feature

ZiliaAJY committed:
Merge pull request #248 from ZiliaAJY/branch-Homepage

Update Home Page

Chinorea committed:
Fix naming convention in UG

Chinorea committed:
Merge pull request #249 from Chinorea/branch-PE-D-Issues

Fix naming convention in UG

Chinorea committed:
Fix naming convention in DG Manual Testing

Commit:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-1/tp

Commit:
Add s into search expenses in command summary of UG

Chinorea committed:
Merge pull request #250 from Chinorea/branch-PE-D-Issues

Fix naming convention in DG Manual Testing

Commit:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-1/tp

Commit:
Merge branch 'master' into branch-addSInSearchExpense

Commit:
Edit formatting errors in DG.

ChuaCleon committed:
Merge pull request #251 from ChuaCleon/branch-addSInSearchExpense

Branch add s in search expense

telferang committed:
Refactor code to use more SLAP

telferang committed:
Update help command

telferang committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-1/tp into branch-updatePPP

# Conflicts:
#	src/main/java/seedu/budgetbuddy/Ui.java

telferang committed:
Merge pull request #252 from telferang/branch-updatePPP

Branch update help command

MayFairMI6 commented:
![](https://ay2425s1-cs2113-w10-1.github.io/tp/diagrams/ExpenseAndIncomeClassDiagram.drawio.png)

In reviewing the class diagram, would it be beneficial to disclose parameters and methods at the class level for the Expense and Income classes? While abstraction simplifies the architectural overview, could the inclusion of specific details regarding how these classes interact through shared parameters and method invocations enhance understanding and facilitate system integration?

Regarding the non-functional requirements, should the dependency on Java 17 be classified as a functional requirement, given that the operability of the project hinges on this version? Does this classification reflect the necessity of this requirement for the software’s operation, distinguishing it from typical non-functional requirements that address system qualities?

In reference to section 5.5.2.1.2 of the document, is it necessary to maintain the second test case under the 'Test Cases' heading if it does not provide additional verification value or distinct scenario coverage? Would its exclusion potentially streamline the testing process and enhance focus on more critical tests?

In reviewing the documentation further, I noticed that the term "Mainstream" is used only once to categorize compatible operating systems. Given its limited use, would it not be more direct and clear to list the compatible operating systems explicitly within the main document? This approach would enhance accessibility and eliminate the need for referencing the Glossary for a single mention, streamlining the document significantly.

On the other hand, the term "Transactions" is effectively used to describe the interactions between financial classes such as Budget, Expense, and Income. This term accurately depicts the financial exchanges or data manipulations that occur within the system, offering a precise illustration of the relationships among these entities. Hence, the use of "Transactions" is appropriate and should be retained as it clearly conveys the intended processes and relationships within the project's framework.
